WriteVHDLSyntax.py

```diff
@@ -1233,6 +1329,46 @@ def writeProcCombination(module, str_ctrl_func, str_ports):
     # FIXME needs fixing to include template parameters for generic proc module writing
     """
     module_str = ""
+
+    if "PC_" in module.inst:
+        module_str += "  " + module.inst + "_mem_reader : entity work.mem_reader\n"
+        module_str += "    generic map (\n"
+        module_str += "      RAM_WIDTH    => 73,\n"
+        module_str += "      NUM_TPAGES    => 4,\n"
+        module_str += "      NAME    => \""+module.inst+"_mem_reader\"\n"
+        module_str += "    )\n"
+        module_str += "    port map (\n"
+        module_str += "      clk    => clk,\n"
+        module_str += "      bx    => PC_bx_in,\n"
+        module_str += "      start => PC_start,\n"
+        module_str += "      enb   => MPAR_"+module.inst[3:]+"in_enb,\n"
+        module_str += "      addra => MPAR_"+module.inst[3:]+"in_V_readaddr,\n"
+        module_str += "      din   => MPAR_"+module.inst[3:]+"in_V_dout,\n"
+        module_str += "      dout  => MPAR_"+module.inst[3:]+"in_V_tpar,\n"
+        module_str += "      valid  => MPAR_"+module.inst[3:]+"in_valid,\n"
+        module_str += "      index  => MPAR_"+module.inst[3:]+"in_trackletindex,\n"
+        module_str += "      nent  => MPAR_"+module.inst[3:]+"in_AV_dout_nent\n"
+        module_str += "    );\n\n"
+
+    if "VMSMER_" in module.inst:
+        module_str += "  " + module.inst + "_mem_reader : entity work.mem_reader\n"
+        module_str += "    generic map (\n"
+        module_str += "      RAM_WIDTH    => 36,\n"
```

Comment:
@aryd Instead of hard-coding this number, can it be gotten from the `module` object? Maybe something like:
```suggestion
        module_str += "      RAM_WIDTH    => " + str(module.upstreams[0].bitwidth) + ",\n"
```

Reply:
This has been fixed.